ludwig/api.py

```diff
@@ -138,75 +138,91 @@ def train(
+            randomly split.
+        :param training_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing training data.
+        :param validation_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing validation data.
+        :param test_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing test data.
+        :param training_set_metadata: (Union[str, dict], default: `None`)
+            metadata JSON file or loaded metadata.
+            Intermediate preprocess structure containing the mappings of the input
+            CSV created the first time a CSV file is used in the same
+            directory with the same name and a '.json' extension.
+        :param data_format: (str, default: `None`) format to interpret data sources. Will be inferred
+               automatically if not specified.
+        :param experiment_name: (str, default: `'experiment'`) name for
+            the experiment.
+        :param model_name: (str, default: `'run'`) name of the model that is
+            being used.
+        :param model_resume_path: (str, default: `None`) resumes training of
+            the model from the path specified. The difference with
+            model_load_path is that also training statistics like the current
+            epoch and the loss and performance so far are also resumed
+            effectively continuing a previously interrupted training process.
         :param skip_save_training_description: (bool, default: `False`) disables
-               saving the description JSON file.
+            saving the description JSON file.
         :param skip_save_training_statistics: (bool, default: `False`) disables
-               saving training statistics JSON file.
+            saving training statistics JSON file.
         :param skip_save_model: (bool, default: `False`) disables
-               saving model weights and hyperparameters each time the model
-               improves. By default Ludwig saves model weights after each epoch
-               the validation metric imrpvoes, but if the model is really big
-               that can be time consuming if you do not want to keep
-               the weights and just find out what performance can a model get
-               with a set of hyperparameters, use this parameter to skip it,
-               but the model will not be loadable later on.
+            saving model weights and hyperparameters each time the model
+            improves. By default Ludwig saves model weights after each epoch
+            the validation metric improves, but if the model is really big
+            that can be time consuming if you do not want to keep
+            the weights and just find out what performance can a model get
+            with a set of hyperparameters, use this parameter to skip it,
+            but the model will not be loadable later on.
```

Comment:
Add at the end: and the returned model will have the weights obtained at the end of training, instead of the weights of the epoch with the best validation performance.

Reply:
done.